Framework/Backend/http/openid.js

```diff
@@ -0,0 +1,84 @@
+/**
+ * @license
+ * Copyright CERN and copyright holders of ALICE O2. This software is
+ * distributed under the terms of the GNU General Public License v3 (GPL
+ * Version 3), copied verbatim in the file "COPYING".
+ *
+ * See http://alice-o2.web.cern.ch/license for full licensing information.
+ *
+ * In applying this license CERN does not waive the privileges and immunities
+ * granted to it by virtue of its status as an Intergovernmental Organization
+ * or submit itself to any jurisdiction.
+*/
+
+const {Issuer, generators} = require('openid-client');
+const assert = require('assert');
+const log = new (require('./../log/Log.js'))('OpenID');
+
+/**
+ * Authenticates and authorises users via OpenID Connect (new CERN SSO).
+ * @author Adam Wegrzynek <adam.wegrzynek@cern.ch>
+ */
+class OpenId {
+  /**
+  * Sets up OpenID Connect based on passed config
+  * @param {object} config - id, secret, welll_known, redirect_uri
+  */
+  constructor(config) {
+    assert(config.id, 'Missing config value: OpenID.id');
+    assert(config.secret, 'Missing config value: OpenID.secret');
+    assert(config.well_known, 'Missing config value: OpenId.well_known');
+    assert(config.redirect_uri, 'Missing config value: OpenId.redirect_uri');
+    this.config = config;
+    this.code_verifier = generators.codeVerifier();
+    this.createIssuer();
```

Comment:
A constructor should only initialize the module and not try to create the issuer. 

This was the same issue we had with the constructor crashing the server.

Reply:
In this case if we can't create issuer we should not start server.